src/osvr/Client/SkeletonRemoteFactory.cpp

```diff
@@ -0,0 +1,152 @@
+/** @file
+    @brief Implementation
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include "SkeletonRemoteFactory.h"
+#include <osvr/Common/ApplyPathNodeVisitor.h>
+#include <osvr/Common/ClientInterface.h>
+#include <osvr/Common/CreateDevice.h>
+#include <osvr/Common/JSONTransformVisitor.h>
+#include <osvr/Common/OriginalSource.h>
+#include <osvr/Common/ProcessArticulationSpec.h>
+#include <osvr/Common/Transform.h>
+#include <osvr/Util/ChannelCountC.h>
+#include <osvr/Util/TreeNode.h>
+#include <osvr/Util/Verbosity.h>
+
+// Library/third-party includes
+#include <boost/algorithm/string/erase.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/optional.hpp>
+#include <boost/variant/get.hpp>
+#include <json/value.h>
+
+// Standard includes
+
+namespace osvr {
+namespace client {
+
+    SkeletonRemoteHandler::SkeletonRemoteHandler(
+        vrpn_ConnectionPtr const &conn, std::string const &deviceName,
+        boost::optional<OSVR_ChannelCount> sensor,
+        common::InterfaceList &ifaces, common::ClientContext *ctx)
+        : m_dev(common::createClientDevice(deviceName, conn)),
```

Comment:
`m_ctx` should be initialized before `m_internals` (see order in SkeletonRemoteFactory.h file.